### PR TITLE
Add support for MinGW

### PR DESCRIPTION
**Pull request by amyspark**
In these cases the build system doesn't add a suffix to yaml-cpp.

Additionally, the `L_tmpnam_s` define has been added only last year in MinGW. 7.3 (the last official release used by Qt 5) doesn't have it, so let's import the define from the official patch.

### CODE REVIEW

File: src/OpenColorIO/Platform.cpp [hunk 1]
```diff
@@ -174,6 +174,12 @@ std::string CreateTempFilename(const std::string & filenameExt)
 
     // Note: Because of security issue, tmpnam could not be used.
 
+    // Note 2: MinGW doesn't define L_tmpnam_s, we need to patch it in.
+    // https://www.mail-archive.com/mingw-w64-public@lists.sourceforge.net/msg17360.html
+#if !defined(L_tmpnam_s)
+#define L_tmpnam_s L_tmpnam
```
Comment (remia) [inline src/OpenColorIO/Platform.cpp:180]: Question: Looking at the link, should we add `#define TMP_MAX_S TMP_MAX` too ?

**Comment by remia** [comment]
Thanks for the fix @amyspark. Looking quickly at other Find*.cmake modules, I wonder if the same update would be required elsewhere? Maybe that could be checked by compiling with `-DOCIO_INSTALL_EXT_PACKAGES=ALL`?

**Comment by amyspark** [comment]
> 
> 
> Thanks for the fix @amyspark. Looking quickly at other Find*.cmake modules, I wonder if the same update would be required elsewhere? Maybe that could be checked by compiling with `-DOCIO_INSTALL_EXT_PACKAGES=ALL`?

I compile using `MISSING` because we already ship OpenEXR and Expat; imath, pystring and yaml-cpp are OK with this patch.

**Comment by amyspark** [comment]
The build error looks spurious:

```
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: CMakeFiles/ociolutimage.dir/main.cpp.o: undefined reference to symbol '_ZN16OpenImageIO_v2_310ParamValue11clear_valueEv'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /usr/local/lib64/libOpenImageIO_Util.so.2.3: error adding symbols: DSO missing from command line
```

Can anyone retry the action?

**Comment by hodoulp** [comment]
The fix is in PR #1478 that I will try to merge soon.

**Review by hodoulp — APPROVED** [review]

**Comment by hodoulp** [comment]
Friendly reminder that the pull request can now be merged.

**Pull request merged**